lib/internal/url.js

```diff
@@ -692,11 +693,14 @@ ObjectDefineProperties(URLSearchParams.prototype, {
  *
  * We use `href` and `protocol` as they are the only properties that are
  * easy to retrieve and calculate due to the lazy nature of the getters.
+ *
+ * We check for auth attribute to distinguish legacy url instance with
+ * WHATWG URL instance.
  * @param {*} self
  * @returns {self is URL}
  */
 function isURL(self) {
-  return Boolean(self?.href && self.protocol);
+  return Boolean(self?.href && self.protocol && !ObjectPrototypeHasOwnProperty(self.auth));
```

Comment:
Probably `self.auth === undefined` is enough but I couldn't choose between the current one.

Reply:
Had to remove this due to `TypeError: Cannot convert undefined or null to object`